src/global_mapper_ros/global_mapper_ros.cc

```diff
@@ -2,77 +2,175 @@
-  map.height = 1;
-
-  int ixyz[3] = {0};
-  int count = 0;
-  for (int i = 0; i < global_mapper_.global_map_.size(); i++) {
-    if (global_mapper_.global_map_[i] > 0.0) {
-      count++;
-      global_mapper_.IndToCoord(i, ixyz);
-      map.points.push_back(pcl::PointXYZ(ixyz[0], ixyz[1], ixyz[2]));
+  std::lock_guard<std::mutex> lock(global_mapper_ptr_->map_mutex_);
+
+  // get occuppied voxels
+  std::vector<std::vector<double> > voxel_vec;
+
+  double xyz[3] = {0.0};
+  std::vector<double> voxel(3, 0.0);
+  for (int i = 0; i < global_mapper_ptr_->voxel_map_ptr_->num_cells; i++) {
+    global_mapper_ptr_->voxel_map_ptr_->indToLoc(i, xyz);
+
+    if (global_mapper_ptr_->voxel_map_ptr_->readValue(xyz) > 0.0) {
+      voxel[0] = xyz[0];
+      voxel[1] = xyz[1];
+      voxel[2] = xyz[2];
+      voxel_vec.push_back(voxel);
     }
   }
-  map.width = map.points.size();
-  ROS_INFO("map points: %i", count);
+  int num_voxels = voxel_vec.size();
+  ROS_INFO("num_voxels: %u", num_voxels);
+
+  visualization_msgs::MarkerArray marker_array;
+
+  visualization_msgs::Marker marker;
+  std::vector<double> rgb(3, 1.0);
+  for (int i = 0; i < num_voxels; ++i) {
+    // create voxel marker
+    marker.header.frame_id = "world";
+    marker.header.stamp = ros::Time::now();
+    marker.id = i;
+    marker.type = visualization_msgs::Marker::CUBE;
+    marker.action = visualization_msgs::Marker::ADD;
+
+    // orientation
+    marker.pose.orientation.x = 0.0;
+    marker.pose.orientation.y = 0.0;
+    marker.pose.orientation.z = 0.0;
+    marker.pose.orientation.w = 1.0;
+
+    // // scale
+    marker.scale.x = global_mapper_ptr_->voxel_map_ptr_->metersPerPixel[0];
+    marker.scale.y = global_mapper_ptr_->voxel_map_ptr_->metersPerPixel[1];
+    marker.scale.z = global_mapper_ptr_->voxel_map_ptr_->metersPerPixel[2];
+
+    // position
+    marker.pose.position.x = voxel_vec[i][0];
+    marker.pose.position.y = voxel_vec[i][1];
+    marker.pose.position.z = voxel_vec[i][2];
+
+    // color
+    rgb = GrayscaleToRGBJet(marker.pose.position.z,
+                            global_mapper_ptr_->voxel_map_ptr_->xyz0[2],
+                            global_mapper_ptr_->voxel_map_ptr_->xyz1[2]);
+    marker.color.r = static_cast<float>(rgb[0]);
+    marker.color.g = static_cast<float>(rgb[1]);
+    marker.color.b = static_cast<float>(rgb[2]);
+    marker.color.a = 1.0f;
+
+    // add voxel marker to marker array
+    marker_array.markers.push_back(marker);
+  }
 
   // publish
-  map_pub_.publish(map);
+  map_pub_.publish(marker_array);
 }
 
-// void GlobalMapperRos::PointCloudCallback(const sensor_msgs::PointCloud2::ConstPtr& cloud_ptr) {
 void GlobalMapperRos::PointCloudCallback(const pcl::PointCloud<pcl::PointXYZ>::ConstPtr& cloud_ptr) {
   // get transform
   const std::string target_frame = "world";
   geometry_msgs::TransformStamped transform_stamped;
   // try to get correct transform
   try {
+    // TODO(jakeware): FIX THIS! WHY THE FUCK DOES THIS NOT WORK?
```

Comment:
Agree, but no combination of --clock, use_sim_time, and roscore and rviz restarts seems to fix it.  Might be something wrong with the timing in the bag.